core/src/Cuda/Kokkos_Cuda_ZeroMemset.hpp

```diff
@@ -36,11 +36,10 @@ struct ZeroMemset<Kokkos::Cuda, View<T, P...>> {
 
   ZeroMemset(const View<T, P...>& dst,
              typename View<T, P...>::const_value_type&) {
-    // FIXME_CUDA_MULTIPLE_DEVICES
+    KOKKOS_IMPL_CUDA_SAFE_CALL(cudaSetDevice(Cuda().cuda_device()));
     KOKKOS_IMPL_CUDA_SAFE_CALL(
-        (Kokkos::Impl::CudaInternal::singleton().cuda_memset_wrapper(
-            dst.data(), 0,
-            dst.size() * sizeof(typename View<T, P...>::value_type))));
+        cudaMemset(dst.data(), 0,
+                   dst.size() * sizeof(typename View<T, P...>::value_type)));
```

Comment:
Wasn't the previous version running on the stream of the default exec space while this one is a blocking call?

Reply:
Daniel pointed out #6187